analyses/npl/09_flood_events_by_municipality.md

```diff
@@ -0,0 +1,300 @@
+We have a file from the RCO that has flood eents only in the municipalities 
+of interest. Here we want to correlate these events with past GloFAS activations.
+
+```python
+from pathlib import Path
+import os
+import sys
+from importlib import reload
+
+import matplotlib as mpl
+import matplotlib.pyplot as plt
+import pandas as pd
+import geopandas as gpd
+import numpy as np
+
+path_mod = f"{Path(os.path.dirname(os.path.realpath(''))).parents[0]}/"
+sys.path.append(path_mod)
+
+from src.indicators.flooding.glofas import utils
+reload(utils)
+
+mpl.rcParams['figure.dpi'] = 200
+pd.options.mode.chained_assignment = None 
+```
+
+```python
+COUNTRY_ISO3 = 'npl'
+
+STATIONS = {
+    'Koshi': 'Chatara',
+    'Karnali': 'Chisapani'
+}
+# Use "_v3" for the GloFAS model v3 locs, or empty string for the original v2 ones
+VERSION_LOC = "_v3"
+
+KARNALI_BASINS = [
+    'Karnali', 'West Rapti', 'Babai'
+]
+KOSHI_WATERSHED = 'Saptakoshi'
+
+DATA_DIR = Path(os.environ["AA_DATA_DIR"]) 
+DATA_DIR_PRIVATE = DATA_DIR / 'private'
+DATA_DIR_PUBLIC = DATA_DIR / 'public'
+RCO_DIR = DATA_DIR_PRIVATE / 'exploration/npl/unrco'
+SHAPEFILE_DIR = DATA_DIR_PUBLIC / 'raw/npl/cod_ab'
+
+
+BASINS_SHAPEFILE = RCO_DIR / 'shapefiles/Major_River_Basins.shp'
+WATERSHED_SHAPEFILE = RCO_DIR / 'shapefiles/Major_watershed.shp'
+ADMIN_SHAPEFILE = SHAPEFILE_DIR / 'npl_admbnda_ocha_20201117/npl_admbnda_nd_20201117_shp.zip'
+#ADMIN2_SHAPEFILE = 'npl_admbnda_districts_nd_20201117.shp'
+ADMIN2_SHAPEFILE = 'npl_admbnda_adm2_nd_20201117.shp'
+
+PAST_EVENTS_FILENAME = RCO_DIR / 'HistoricalReportedIncident_Municipality Level_AAPilot.xlsx'
+
+GLOFAS_DIR = DATA_DIR / "public/exploration/npl/glofas"
+GLOFAS_RP_FILENAME = GLOFAS_DIR / "glofas_return_period_values.xlsx"
+
+MUNICIPALITIES_OUTPUT_GEOPACKAGE = RCO_DIR / 'shapefiles/municipalities_of_interest.gpkg'
+```
+
+### Read in and clean data
+
+Read in GloFAS data
+
+```python
+ds_glofas_reanalysis = utils.get_glofas_reanalysis(
+    country_iso3=COUNTRY_ISO3)
+df_return_period =  pd.read_excel(GLOFAS_RP_FILENAME, index_col='rp')
+```
+
+Read in events and clean
+
+```python
+df_events = (pd.read_excel(PAST_EVENTS_FILENAME, index_col='S.No.')
+             .rename(columns={'Mun_ETHOS_Code': 'pcode'})
+            )
+# Drop events with no date and convert date column
+df_events = df_events.loc[df_events['Year'] > 0]
```

Comment:
How many events is this dropping? Might be good to keep track of that as a caveat in any derived results.

Reply:
Ah actually it's not dropping any, this line was a remnant from the old events file. I've removed it, good catch!